### PR TITLE
Tool 1129 Add Scheme metadata for the uploaded XcArchive

### PR DESCRIPTION
**Pull request by BirmacherAkos**
New metadata for the XcArchive
- scheme

---

Connected website PR
https://github.com/bitrise-io/bitrise-website/pull/3423

### CODE REVIEW

File: uploaders/xcarchiveuploader.go [hunk 1]
```diff
@@ -29,22 +30,29 @@ func DeployXcarchive(pth, buildURL, token string) error {
 		return nil
 	}
 
-	infoPlistPth, err := xcarchive.GetEmbeddedInfoPlistPath(filepath.Join(unzippedPth, pathutil.GetFileName(pth)))
+	appInfoPlistPth, err := xcarchive.GetEmbeddedInfoPlistPath(filepath.Join(unzippedPth, pathutil.GetFileName(pth)))
```
Comment (godrei) [inline uploaders/xcarchiveuploader.go]: `xcarchive.GetEmbeddedInfoPlistPath(archivePth)`

File: uploaders/xcarchiveuploader.go [hunk 2]
```diff
@@ -29,22 +30,29 @@ func DeployXcarchive(pth, buildURL, token string) error {
...
 	if err != nil {
 		return fmt.Errorf("failed to unwrap Info.plist from xcarchive, error: %s", err)
 	}
 
-	infoPlistData, err := plistutil.NewPlistDataFromFile(infoPlistPth)
+	appInfoPlistData, err := plistutil.NewPlistDataFromFile(appInfoPlistPth)
```
Comment (godrei) [inline uploaders/xcarchiveuploader.go]: NOTE: this functionality could be added to xcarchive package

**Comment by BirmacherAkos** [comment]
I've updated the XcArchive parsing logic:
- The go-xcode package has a logic for pasring iOSArvhive or macOSArvhive. => Using that instead.

File: uploaders/xcarchiveuploader.go (hunk 2 above)
Comment (BirmacherAkos) [reply uploaders/xcarchiveuploader.go]: We even have better:
https://github.com/bitrise-io/go-xcode/blob/master/xcarchive/ios.go#L188

File: uploaders/xcarchiveuploader.go [hunk 3]
```diff
@@ -21,38 +20,25 @@ func DeployXcarchive(pth, buildURL, token string) error {
...
-	appInfoPlistPth, err := xcarchive.GetEmbeddedInfoPlistPath(filepath.Join(unzippedPth, pathutil.GetFileName(pth)))
-	if err != nil {
-		return fmt.Errorf("failed to unwrap Info.plist from xcarchive, error: %s", err)
-	}
-
-	appInfoPlistData, err := plistutil.NewPlistDataFromFile(appInfoPlistPth)
-	if err != nil {
-		return fmt.Errorf("failed to parse Info.plist (%s), error: %s", appInfoPlistPth, err)
+	} else if isMacos {
+		return nil // MacOS project is not supported, so won't be deployed.
```
Comment (godrei) [inline uploaders/xcarchiveuploader.go]: pls log a warning about it

**Pull request merged**